crmsh/ui_cluster.py

```diff
@@ -231,24 +232,29 @@ def looks_like_hostnames(lst):
                                    help="Configure corosync with second heartbeat line")
         network_group.add_argument("-I", "--ipv6", action="store_true", dest="ipv6",
                                    help="Configure corosync use IPv6")
-        network_group.add_argument("--qnetd-hostname",
+
+        qdevice_group = parser.add_argument_group("QDevice configuration", "Options for configuring QDevice and QNetd.")
+        qdevice_group.add_argument("--qnetd-hostname",
                                    dest="qdevice", metavar="HOST",
                                    help="HOST or IP of the QNetd server to be used")
-        network_group.add_argument("--qdevice-port",
+        qdevice_group.add_argument("--qdevice-port",
                                    dest="qdevice_port", metavar="PORT", type=int, default=5403,
                                    help="TCP PORT of QNetd server(default:5403)")
-        network_group.add_argument("--qdevice-algo",
+        qdevice_group.add_argument("--qdevice-algo",
                                    dest="qdevice_algo", metavar="ALGORITHM", default="ffsplit",
                                    help="QNetd decision ALGORITHM(ffsplit/lms, default:ffsplit)")
-        network_group.add_argument("--qdevice-tie-breaker",
+        qdevice_group.add_argument("--qdevice-tie-breaker",
                                    dest="qdevice_tie_breaker", metavar="TIE_BREAKER", default="lowest",
                                    help="QNetd TIE_BREAKER(lowest/highest/valid_node_id, default:lowest)")
-        network_group.add_argument("--qdevice-tls",
+        qdevice_group.add_argument("--qdevice-tls",
                                    dest="qdevice_tls", metavar="TLS", default="on",
                                    help="Whether using TLS on QDevice/QNetd(on/off/required, default:on)")
-        network_group.add_argument("--qdevice-heuristics",
+        qdevice_group.add_argument("--qdevice-heuristics",
                                    dest="qdevice_heuristics", metavar="COMMAND",
                                    help="COMMAND to run with absolute path. For multiple commands, use \";\" to separate(details about heuristics can see man 8 corosync-qdevice)")
+        qdevice_group.add_argument("--qdevice-heuristics-mode",
```

Comment:
You say that `sync` is the default value. Why don't you put it in the `default` parameter?
I have seen in other piece of code that you do this manually. Why don't let the parser do it natively?

Reply:
I did it before, but change it into this line https://github.com/liangxin1300/crmsh/blob/34bfc3b74b759ea6bb92f855e0797f5fc22aac76/crmsh/corosync.py#L490
Cause if not specify `--qdevice-heuristics`, `--qdevice-heuristics-mode` with a default value is useless

Reply:
Ok, now I understand what do you mean.
So, in my opinion I would change the location where you assign the default value. And put after the piece of code ```
            if options.qdevice_heuristics_mode and not options.qdevice_heuristics:
                parser.error("Option --qdevice-heuristics is required if want to configure heuristics mode")
            options.qdevice_heuristics_mode = options.qdevice_heuristics_mode or "sync"
```
This way, you have all the logic of the parameters in the parser, where it should belong to

Reply:
OK, changed